sky/backends/cloud_vm_ray_backend.py

```diff
@@ -2864,16 +2869,29 @@ def _get_zone(runner):
                     # launched in different zones (legacy clusters before
                     # #1700), leave the zone field of handle.launched_resources
                     # to None.
-            self._update_after_cluster_provisioned(handle, task,
-                                                   prev_cluster_status, ip_list,
-                                                   ssh_port_list, lock_path)
+            self._update_after_cluster_provisioned(
+                handle, task, prev_cluster_status, ip_list, ssh_port_list,
+                to_provision_config.ports_to_open, lock_path)
```

Comment:
We could, but as I said before, we need the `to_provision` resources to contain the newest ports... So I put the calculation for `ports_to_open` into `_check_existing_cluster` too to avoid calculate it twice 🤔